ciri/conf/getopt.c

```diff
@@ -0,0 +1,34 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include "ciri/conf/provider.h"
+#include "ciri/conf/config_args.h"
+
+ciri_config_t* ciri_conf_init() {
+  ciri_config_t *config = (ciri_config_t* )malloc(sizeof(ciri_config_t));
```

Comment:
If we put an instance of `ciri_config_t` in the core, we can get rid of this allocation, by the way it was leaking.